core/modules/css/CssAccess.cc

```diff
@@ -53,31 +68,106 @@ namespace qserv {
-}
+CssAccess::createFromConfig(std::map<std::string, std::string> const& config,
+                            std::string const& emptyChunkPath,
+                            bool readOnly) {
+    LOGF(_log, LOG_LVL_DEBUG, "Create CSS instance from config map");
+    auto iter = config.find("technology");
+    if (iter == config.end()) {
+        LOGF(_log, LOG_LVL_DEBUG, "\"technology\" does not exist in configuration map");
+        throw ConfigError("\"technology\" does not exist in configuration map");
+    } else if (iter->second == "mem") {
+        // optional data or file keys
+        auto iterData = config.find("data");
+        auto iterFile = config.find("file");
+        if (iterData != config.end() and iterFile != config.end() ) {
+            LOGF(_log, LOG_LVL_DEBUG, "\"data\"  and \"file\" keys are mutually exclusive");
+            throw ConfigError("\"data\"  and \"file\" keys are mutually exclusive");
+        }
+        if (iterData != config.end()) {
+            // data is in a string
+            return createFromData(iterData->second, emptyChunkPath, readOnly);
+        } else if (iterFile != config.end()) {
+            // read data from file
+            std::ifstream f(iterFile->second);
+            if(f.fail()) {
+                LOGF(_log, LOG_LVL_DEBUG, "failed to open data file %s" % iterFile->second);
+                throw ConfigError("failed to open data file " + iterFile->second);
+            }
+            LOGF(_log, LOG_LVL_DEBUG,
+                 "Create CSS instance with memory store from data file %s" % iterFile->second);
+            auto kvi = std::make_shared<KvInterfaceImplMem>(f, readOnly);
+            return std::shared_ptr<CssAccess>(new CssAccess(kvi,
+                                                            std::make_shared<EmptyChunks>(emptyChunkPath)));
+        } else {
+            // no initial data
+            LOGF(_log, LOG_LVL_DEBUG, "Create CSS instance with empty memory store");
+            auto kvi = std::make_shared<KvInterfaceImplMem>(readOnly);
+            return std::shared_ptr<CssAccess>(new CssAccess(kvi,
+                                                            std::make_shared<EmptyChunks>(emptyChunkPath)));
+        }
+    } else if (iter->second == "mysql") {
+        // extract all optional values from map
+        mysql::MySqlConfig mysqlConfig;
+        mysqlConfig.hostname = _mapGet(config, "hostname");
+        mysqlConfig.username = _mapGet(config, "username");
+        mysqlConfig.password = _mapGet(config, "password");
+        mysqlConfig.dbName = _mapGet(config, "database");
+        mysqlConfig.socket = _mapGet(config, "socket");
+        auto portStr = _mapGet(config, "port", "");
+        if (portStr.empty()) portStr = "0";
+        try {
+            // tried to use std::stoi() here but it returns OK for strings like "0xFSCK"
+            mysqlConfig.port = boost::lexical_cast<unsigned>(portStr);
```

Comment:
I think the way to detect it in call to `stoi()` is something like this:

    size_t pos;
    auto value = std::stoi(str, &pos);
    if (pos != str.size()) {
        // error
    }

Which I do not particularly like, it's not much easier to use than good old `strtol()` (but maybe I'm too used to boost stuff).